### PR TITLE
[CORE-1766] Conditionally show/hide unit questions based on form metadata

### PR DESCRIPTION
**Pull request by MarkDKoslow**
## Jira Issue
[CORE-1766]

## Description
Issue with last PR was that I needed to relax the isEqual check to isMatch, because the form metadata is now an object with two keys "showBuildWeekQuestions" and "showUnitQuestions"

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

[CORE-1766]: https://lambdaschool.atlassian.net/browse/CORE-1766

### CODE REVIEW

File: utils/is-question-or-question-group-visible/index.ts [hunk 1]
```diff
@@ -54,7 +55,7 @@ function checkQuestionOrQuestionGroupVisibility(questionOrQuestionGroup: Questio
       );
     }
 
-    return isEqual(parentMetadata, questionOrQuestionGroup.visibleIf.answer);
+    return !!parentMetadata && isMatch(parentMetadata, questionOrQuestionGroup.visibleIf.answer);
```
Comment (mellogarrett) [inline utils/is-question-or-question-group-visible/index.ts]: Do you need `!!parentMetadata`? If you have to have something there, will just `return parentMetadata &&...` work?

Comment (MarkDKoslow) [reply utils/is-question-or-question-group-visible/index.ts]: I think that would also work. It looks like all the other code paths for this function return a boolean, so I mostly added that to keep it consistent. I also just prefer it stylistically, but don't have a strong preference here

Comment (mellogarrett) [reply utils/is-question-or-question-group-visible/index.ts]: Do you prefer it over the more explicit `parentMetadata !== undefined`? That would be more clear IMO. 

Doesn't matter much either way, just being picky :)

Comment (MarkDKoslow) [reply utils/is-question-or-question-group-visible/index.ts]: Good call, I like `parentMetadata !== undefined` more. Updating 👍

**Pull request merged**